src/NuGet.Clients/PackageManagement.UI/PackageLoader.cs

```diff
@@ -283,6 +289,107 @@ private IEnumerable<string> GetSupportedFrameworks()
             };
         }
 
+        /// <summary>
+        /// Returns the list of packages that are consolidatable, i.e. different versions of the
+        /// package are installed.
+        /// </summary>
+        /// <returns></returns>
+        public async Task<ICollection<PackageIdentity>> GetConsolidatablePackagesAsync(
+            CancellationToken token)
+        {
+            if (_projects.Length <= 1)
+            {
+                return new List<PackageIdentity>();
+            }
+
+            // the key of the dictionary is the package id, and the value is
+            // versions installed.
+            var packages = new Dictionary<string, HashSet<NuGetVersion>>(
+                StringComparer.OrdinalIgnoreCase);
+
+            foreach (var project in _projects)
+            {
+                foreach (var package in (await project.GetInstalledPackagesAsync(token)))
+                {
+                    HashSet<NuGetVersion> versions;
+                    if (!packages.TryGetValue(package.PackageIdentity.Id, out versions))
+                    {
+                        versions = new HashSet<NuGetVersion>();
+                        packages[package.PackageIdentity.Id] = versions;
+                    }
+
+                    versions.Add(package.PackageIdentity.Version);
+                }
+            }
+
+            var consolidatablePackages = packages
+                .Where(p => p.Value.Count >= 2)
+                .Select(p => new PackageIdentity(p.Key, p.Value.Max()))
+                .ToList();
+
+            return consolidatablePackages;
+        }
+
+        private async Task<SearchResult> SearchConsolidateAsync(
+            int startIndex, 
+            CancellationToken cancellationToken)
+        {
+            var installedPackages = (await GetConsolidatablePackagesAsync(token: cancellationToken))
```

Comment:
rename to packagesNeedingConsolidation or something like that. Installedpackages is misleading